x-pack/test/security_solution_cypress/cypress/tasks/prebuilt_rules/setup_rule_upgrades.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  RuleResponse,
+  RuleSignatureId,
+} from '@kbn/security-solution-plugin/common/api/detection_engine';
+import { patchRule } from '../api_calls/rules';
+import {
+  SAMPLE_PREBUILT_RULE,
+  installPrebuiltRuleAssets,
+  installSpecificPrebuiltRulesRequest,
+} from '../api_calls/prebuilt_rules';
+import { deletePrebuiltRulesAssets } from '../api_calls/common';
+
+interface SetUpRulesParams {
+  currentRuleAssets: Array<typeof SAMPLE_PREBUILT_RULE>;
+  newRuleAssets: Array<typeof SAMPLE_PREBUILT_RULE>;
+  rulePatches: Array<{ rule_id: RuleSignatureId } & Partial<RuleResponse>>;
+  cleanUpCurrentRuleAssets?: boolean;
```

Comment:
nit: This would be used for removing base versions for testing right? I think something like `removeCurrentRuleAssets` or `deleteCurrentRuleAssets` might be a better descriptor.